CHANGELOG.md

```diff
@@ -64,6 +64,8 @@ Change Log
     `DerivationTool.deriveUnifiedSpendingKey` instead).
   - `DerivationTool.deriveViewingKey` (use
   - `DerivationTool.deriveUnifiedFullViewingKey` instead).
+  - `DerivationTool.deriveTransparentAddress` (use
+    `Synchronizer.getLegacyTransparentAddress` instead).
```

Comment:
Should we remove any of the other transparent address derivation methods? This one was the most confusing, and also happens to be the only one that wasn't altered during UA integration. The other old ones removed below were because we replaced them with UA-compatible equivalents, but those still allow users to generate arbitrary taddrs that don't necessarily appear in the UAs the SDK will generate for accounts (if the UAs contain Sapling, which they do currently).

Reply:
It seems like we eventually want to refactor this to simplify it.  Whether that time is now is later is ¯\_(ツ)_/¯ 

Given that the DerivationTool has `deriveUnifiedAddress`, would it make more sense for the remaining methods to take the UnifiedAddress as a parameter and derive the others from it?

Would that line of thought lead us towards having a UnifiedAddress object with accessor methods for getSaplingAddress and getTransparentAddress?

Reply:
We already have those methods on `RustBackend`: `getTransparentReceiver` and `getSaplingReceiver`. If those would make more sense moved to `DerivationTool`, we can do that.

Reply:
Re: the others, the question is whether there remains e.g. a use case for deriving a transparent address from a transparent account private key. At the time I ported the existing methods to be compatible with ZIP 316, we weren't planning to pass around an encoded USK. Now that we are, it's likely that these methods simply aren't necessary now, so we could just remove them.

Reply:
I vote remove and simplify the public API.

Reply:
> Re: the others, the question is whether there remains e.g. a use case for deriving a transparent address from a transparent account private key.

no use for this since on the SDK this derivation was a workaround and transparent priv keys were never stored just passed through the ffi

Reply:
Okay, I will do this.